### PR TITLE
added a nlp section section

### LINKED ISSUE #1919: [ADD] <AI/Natural Language Processing>
### Category Name

AI

### Subcategory Name

Natual Language Processing

### Additional Context

In the AI category a subcategory must be there of Natural Language processing which is missing. I want to add it. Please assign it to me.

### Checklist

- [X] I have checked the existing [issues](https://github.com/rupali-codes/LinksHub/issues?q=is%3Aissue+)
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I have checked that this category does not exist
- [X] I have checked that this subcategory does not exist
- [X] I am willing to work on this issue (optional)
- [X] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by Arnabhit**
Fixes Issue
Closes  #1919 

## Changes proposed

I Have added a subcategory Natural Language Processing under the category AI. 

## Screenshots

Screenshot (63)

Screenshot (62)

### CODE REVIEW

File: database/artificial_intelligence/natural-language-processing.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[
+    {
+        "name": "Applied Natural Language Processing",
+        "description": "This course starts with the basics of text processing including basic pre-processing, spelling correction, language modeling, Part-of-Speech tagging, Constituency and Dependency Parsing, Lexical Semantics, distributional Semantics and topic models.",
```
Comment (CBID2) [inline database/artificial_intelligence/natural-language-processing.json:4]:
```suggestion
        "description": "This course teaches the basics of text processing, including  pre-processing, spelling correction, language modeling, Part-of-Speech tagging, Constituency and Dependency Parsing, Lexical Semantics, distributional Semantics, and topic models.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Made some suggestions @Arnabhit

File: database/artificial_intelligence/natural-language-processing.json (hunk 1 above)
Comment (Arnabhit) [reply database/artificial_intelligence/natural-language-processing.json]: i have done it .Please review it

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
Shorten the subcategory name to nlp, people who know will understand it.

image

Also, there should be no `-` between the name which happened due to this

image

**Comment by Arnabhit** [comment]
ok i will do it

**Comment by Anmol-Baranwal** [comment]
@Arnabhit 
Please take your time, no need to rush. We will review whenever you have made the changes.

**Comment by Arnabhit** [comment]
ok thank you

**Comment by Arnabhit** [comment]
done

File: database/artificial_intelligence/NLP.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        "name": "Datacamp Tutorials",
+        "description": "Provides NLP tutorials for Machine Learning Beginners",
+        "url": "https://www.datacamp.com/blog/what-is-natural-language-processing",
+        "category": "ai",
+        "subcategory": "natural-language-processing"
+      },
+      {
+        "name": "fast.ai NLP course",
+        "description": " Fast.ai's Code-First Introduction to Natural Language Processing for beginner's.",
```
Comment (CBID2) [inline database/artificial_intelligence/NLP.json:18]:
```suggestion
        "description": " Fast.ai's Code-First Introduction to Natural Language Processing for beginners.",
```

File: database/artificial_intelligence/NLP.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        "url": "https://www.fast.ai/posts/2019-07-08-fastai-nlp.html",
+        "category": "ai",
+        "subcategory": "natural-language-processing"
+      },
+      {
+        "name": "Free Code Camp",
```
Comment (CBID2) [inline database/artificial_intelligence/NLP.json:24]:
```suggestion
        "name": "freeCodeCamp",
```

File: database/artificial_intelligence/NLP.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        "name": "Applied Natural Language Processing",
+        "description": "This course teaches  the basics of text processing including basic pre-processing, spelling correction, language modeling, Part-of-Speech tagging, Constituency and Dependency Parsing, Lexical Semantics, distributional Semantics and topic models.",
+        "url": "https://archive.nptel.ac.in/courses/106/106/106106211/",
+        "category": "ai",
+        "subcategory": "natural-language-processing"
+    },
+    {
+        "name": "Datacamp Tutorials",
+        "description": "Provides NLP tutorials for Machine Learning Beginners",
```
Comment (CBID2) [inline database/artificial_intelligence/NLP.json:11]:
```suggestion
        "description": "Provides NLP tutorials for Machine Learning beginners",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @Arnabhit! :) I pointed out a proofreading error in my inline comment. In the future, consider running your descriptions through Grammarly or Spellcheck before making a pull request. Other than that, you're off to a great start! :)

**Comment by Arnabhit** [comment]
done 🥲🥲

**Review by Anmol-Baranwal — COMMENTED** [review]
@rupali-codes 
Can you check why we can't access the description of it?

image

Other than that, everything seems fine.

**Comment by Arnabhit** [comment]
I am also not able to click the description icon. I am not able to do it. Can you plz help

**Comment by Anmol-Baranwal** [comment]
@Arnabhit 
Just add a description in `TopBar/CategoryDescriptions.ts`. We will resolve the bug internally.

**Comment by Arnabhit** [comment]
added the description

**Review by CBID2 — APPROVED** [review]
Looks good

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks good to me.

**Comment by rupali-codes** [comment]
> @rupali-codes Can you check why we can't access the description of it?
> 
> image
> 
> Other than that, everything seems fine.

it was the name, it should be as per the URL but without dash or anything

**Review by rupali-codes — APPROVED** [review]
lgtm

**Pull request merged**